src/app/Main/Body/QueryPage/Drawer/Filter/Filter-SelectorList.component.tsx

```diff
@@ -26,19 +28,29 @@ export function FilterSelectorListComponent(
     ): void => onChange(selectedOption, keyName);
 
     const elements: JSX.Element[] = [];
-    mainFilterAttributes.forEach((mainFilter) => {
-        elements.push(
-            SelectorListSelectorComponent({
-                dataUniqueValues,
-                selectedFilter,
-                filterAttribute: mainFilter,
-                onChange: handleChange,
-            })
-        );
+    mainFilter.forEach((filter) => {
+        if ( _.includes(filtersToDisplay, filter)) {
+            elements.push(
+                SelectorListSelectorComponent({
+                    hide: false,

```

Comment:
Why not just ignore the SelectorListSelectorComponent completely if the filter is disabled? Than you can remove the hide prop.

Reply:
Yes, that was my first thought too. However, react then complains that it has more components than in the previous render.

Reply:
Can you compile it in release mode and check  if the error is still there? This might be the known problem that the components change during two render cycles and something with the data flow is wrong. Perhaps you must compose the components in an effect listening to the filtersToDisplay changes.

Reply:
Even after I put all the logic from the dialog into the container, it doesn't work. I don't know how to fix it.

Reply:
I will look into it if the ticket is merged into develop